viewing_party/party.py

```diff
@@ -1,23 +1,213 @@
+        list_of_watched_genre_counts.append(genre_count_value)
+
+    #Determines the highest genre count value
+    max_watched_genre_count=max(list_of_watched_genre_counts)
+
+    #Returns a genre from the genre:counter dictionary with a counter value that is the same as the highest genre count value
+    for genre,genre_count_value in genre_counter_dict.items():
+        if genre_count_value == max_watched_genre_count:
+            return genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(single_user_data):
+    list_of_user_movie_info=single_user_data["watched"]
+    list_of_frnds_watched_dicts=single_user_data["friends"]
+    list_of_user_movie_titles=[]
+    list_of_friends_movie_titles=[]
+    list_of_unique_user_movie_info=[]
+    
+    #Adds user's watched movie titles to a list
+    for user_movie_info in list_of_user_movie_info:
+        list_of_user_movie_titles.append(user_movie_info["title"])
+ 
+    #Adds friends' watched movie titles to a list
+    for frnds_watched_movie_info in list_of_frnds_watched_dicts:
+        for frnds_watched_movie_titles in frnds_watched_movie_info["watched"]:
+            list_of_friends_movie_titles.append(frnds_watched_movie_titles["title"])
+
+    #Uses sets data structure and difference method to add user's unique move titles that only the user has seen to a list
+    list_of_unique_user_movie_titles=set(list_of_user_movie_titles).difference(set(list_of_friends_movie_titles))
+    
+    #Adds movie info from a list of the user's watched movie info to a new list if the movie title is in the list of the user's unique movie titles
+    for user_movie_info in list_of_user_movie_info:
+        if user_movie_info["title"] in list_of_unique_user_movie_titles:
+            list_of_unique_user_movie_info.append(user_movie_info)
+    return list_of_unique_user_movie_info
+
+def get_friends_unique_watched(single_user_data):
+    list_of_user_movie_info=single_user_data["watched"]
+    list_of_frnds_watched_dicts=single_user_data["friends"]
+    list_of_user_movie_titles=[]
+    list_of_friends_movie_titles=[]
+    list_of_unique_frnds_movie_info=[]
+
+    #Adds user's watched movie titles to a list
+    for user_movie_info in list_of_user_movie_info:
+        list_of_user_movie_titles.append(user_movie_info["title"])
+
+    #Adds friends' watched movie titles to a list
+    for frnds_watched_movie_info in list_of_frnds_watched_dicts:
+        for frnds_watched_movie_titles in frnds_watched_movie_info["watched"]:
+            list_of_friends_movie_titles.append(frnds_watched_movie_titles["title"])
+    
+    #Uses sets data structure and difference method to add friends' unique move titles that at least one friend has seen to a list
+    list_of_unique_frnds_movie_titles=set(list_of_friends_movie_titles).difference(set(list_of_user_movie_titles))

```

Comment:
hmm where have we seen this code already? maybe we could turn this into a helper function that we call upon inside this function.